### PR TITLE
Unnecessary blank area is created when scrolling down while keyboard is open #19642

### PR DESCRIPTION
**Pull request by lukemorawski**
### Details
Prevent user from scrolling away the message composer and create an odd blank area while virtual keyboard is open on mobile Safari. This is done be scrolling back to the optimal position.

https://github.com/Expensify/App/assets/134388952/b1b535ac-1ce6-43f2-af8e-bc0d0bb57059

### Fixed Issues

$ https://github.com/Expensify/App/issues/19642
PROPOSAL: https://github.com/Expensify/App/issues/19642#issuecomment-1584359639

### Tests

- [ ] Verify that no errors appear in the JS console

### QA Steps

- [ ] Verify that no errors appear in the JS console
- [ ] Open the app in mobile Safari
- [ ] Go to Report Page
- [ ] Tap on message composer component to open the virtual keyboard
- [ ] Scroll the message thread component down (swipe up) to reveal the hidden space under the viewport
- [ ] Release the tap (touchend) - the viewport should smoothly scrollback to it's optimal position hiding that unnecessary space
- [ ] Verify on other mobile and desktop browsers that everything works as expected
- [ ] Verify on native platforms that everything works as expected

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
        - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

SCR-20230622-ifde

</details>

<details>
<summary>Mobile Web - Chrome</summary>

Screenshot_20230622_085040

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/134388952/b1b535ac-1ce6-43f2-af8e-bc0d0bb57059

</details>

<details>
<summary>Desktop</summary>

SCR-20230622-iibt

</details>

<details>
<summary>iOS</summary>

Simulator Screenshot - iPhone 14 - 2023-06-22 at 08 57 12

</details>

<details>
<summary>Android</summary>

Screenshot_1687417357

</details>

### CODE REVIEW

**Comment by lukemorawski** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by lukemorawski** [comment]
recheck

File: src/pages/home/ReportScreen.js
Comment (Santhosh-Sellavel) [inline src/pages/home/ReportScreen.js:3]: There are a few unintended format changes please revert

File: src/components/withBlockViewportScroll/index.js
Comment (Santhosh-Sellavel) [inline src/components/withBlockViewportScroll/index.js:8]: More Comments for the could be added about the component

File: src/components/withBlockViewportScroll/index.js
Comment (Santhosh-Sellavel) [inline src/components/withBlockViewportScroll/index.js:11]: no shortened variable names, and keep meaningful names

**Review by Santhosh-Sellavel — CHANGES_REQUESTED** [review]
Please run through the author checklist once again and make sure everything is met, thanks!

File: src/pages/home/ReportScreen.js
Comment (lukemorawski) [reply src/pages/home/ReportScreen.js]: fixed

File: src/components/withBlockViewportScroll/index.js
Comment (lukemorawski) [reply src/components/withBlockViewportScroll/index.js]: renamed for more clarity

**Comment by Santhosh-Sellavel** [comment]
Resolve conflicts

**Comment by Santhosh-Sellavel** [comment]
@lukemorawski bump!

**Comment by lukemorawski** [comment]
conflict resolved

**Comment by Santhosh-Sellavel** [comment]
@lukemorawski Please tag someone here, so we can follow-up with them if any help is needed, thanks!

**Comment by Santhosh-Sellavel** [comment]
I've played around with the new build. I like the new item, where we keep main the composer in the view by scrolling to it. But it looks glitchy on/off while scrolling to reposition or while the keyboard closes.  @thienlnam Can you have a look at play around with build, and let me know your thoughts, thanks!

**Comment by thienlnam** [comment]
I'm running into issues with my simulator not even opening the keyboard 😵‍💫 Going to be out for the 4th, but will check back in later

Screenshot 2023-07-03 at 4 57 24 PM

**Comment by Santhosh-Sellavel** [comment]
@thienlnam Check `Toggle Software Keyboard` it will make the soft keyboard active. But the moment you start interacting with the app using the hardware keyboard it will disappear again!
Screenshot 2023-07-04 at 10 42 06 PM

**Comment by Santhosh-Sellavel** [comment]
@thienlnam any update?

Also @lukemorawski is OOO can you ask someone else from CallStack to takeover the PR?

**Comment by thienlnam** [comment]
Thanks for the bump, this slipped from my radar - will try to get around to testing this later today

**Comment by Santhosh-Sellavel** [comment]
Okay then let's wait to resolve conflicts and do a final review & testing!

**Comment by lukemorawski** [comment]
Hey all, I'll be resolving the conflicts today.

**Comment by lukemorawski** [comment]
Conflicts resolved

**Comment by Santhosh-Sellavel** [comment]
@lukemorawski Solve lint errors

**Comment by lukemorawski** [comment]
fixed the function declaration linting error

**Comment by lukemorawski** [comment]
fixed next linting error. Weird stuff cause they don't popup locally.

**Comment by Santhosh-Sellavel** [comment]
#### Before
https://github.com/Expensify/App/assets/85645967/fee6336d-5164-4f42-84cd-f4b6bca36c62


#### Now
https://github.com/Expensify/App/assets/85645967/1f8f3e29-3e84-49ad-a5de-ac47e9f31996

**Comment by Santhosh-Sellavel** [comment]
[]()https://github.com/Expensify/App/assets/85645967/ad392e4d-de92-47ef-b456-4f4b43af3d8a

**Comment by Santhosh-Sellavel** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos

https://github.com/Expensify/App/pull/21298#issuecomment-1646950227

<details>
<summary>Mobile Web - Chrome & Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/85645967/45ce59e3-e9c7-4b13-b979-f24a20a266af

</details>

<details>
<summary> Desktop & Web</summary>

https://github.com/Expensify/App/assets/85645967/cbf4785a-bd9f-46ba-bd43-dc058c7a760d

</details>

<details>
<summary>iOS & Android</summary>

https://github.com/Expensify/App/assets/85645967/77e8dff1-859f-4f04-ab90-028684633863

</details>

**Comment by lukemorawski** [comment]
Fixed the keyboard dismissing bug on scrolling

**Comment by Santhosh-Sellavel** [comment]
Still, the keyboard gets dismissed for a single scroll

https://github.com/Expensify/App/assets/85645967/0cd5c1b6-1ebe-46af-ab47-5abbc12b37d4

**Comment by lukemorawski** [comment]
Will investigate that today

**Comment by Santhosh-Sellavel** [comment]
Looks better will continue testing and update tomorrow!

File: src/components/SwipeableView/index.js
Comment (Santhosh-Sellavel) [inline src/components/SwipeableView/index.js:22]: We should use destructuring to assign default values for props https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#destructuring

**Review by Santhosh-Sellavel — CHANGES_REQUESTED** [review]

**Comment by lukemorawski** [comment]
@Santhosh-Sellavel any updates on that? can I help you with anything?

**Comment by Santhosh-Sellavel** [comment]
Sorry I missed this somehow. Let's get to it  later today

**Comment by Santhosh-Sellavel** [comment]
@lukemorawski I'm not sure what went wrong. After merging the main this works weirdly, not sure where it got broken

https://github.com/Expensify/App/assets/85645967/e632569d-64ff-4521-b408-6aae23ff7118

**Comment by lukemorawski** [comment]
right, taking a look at it now

**Comment by lukemorawski** [comment]
I can't reproduce it on a physical device, but it does happen sometimes (when you first open up a chat) on a simulator. 

here's a clip from a physical device
https://github.com/Expensify/App/assets/134388952/53429dfe-9c4d-48de-959d-cb67af5f83d4

**Comment by Santhosh-Sellavel** [comment]
@lukemorawski It happens for me on a physical device too

https://github.com/Expensify/App/assets/85645967/fca33c7d-2756-4332-aeec-4d71b4cb2647

**Review by Santhosh-Sellavel — CHANGES_REQUESTED** [review]
[This](https://github.com/Expensify/App/pull/21298#issuecomment-1673935897) happens to me on [real device](https://github.com/Expensify/App/pull/21298#issuecomment-1678152159) too

**Comment by lukemorawski** [comment]
@Santhosh-Sellavel I tested it again on two separate iPhones on every chat I have. With restarting safari, cleaning cache, etc. I cannot recreate that bug. It happens to me in a Simulator though, but very randomly without any clear correlation.
On a physical device it happened only when I'm running the app without the changes from this PR. But also random.
Maybe we could create a separate ticket for that issue, as it's highly unlikely that this PR caused this bug. I would say it's more like a general Safari input field handling bug (I have seen this behaviour across  multiple web apps). What you say?

**Comment by thienlnam** [comment]
Yeah agreed, we can create a new ticket - but I'm not sure if we should merge this PR without first addressing that problem. If it's easier, we can HOLD this for that new ticket

**Comment by lukemorawski** [comment]
@Santhosh-Sellavel cool, let's separate the two tickets and hold this one. I can volunteer to work on that new ticket, since I'm in the zone already.
I also might have a clue why this bug doesn't occur on my iPhones - I'm running a beta of iOS 17 and my simulator is running v16. There's a chance that apple fixed something in Safari, though I can't confirm that for 100%.

**Comment by lukemorawski** [comment]
@thienlnam will resolve those conflicts today

**Comment by lukemorawski** [comment]
@thienlnam any update on this?

**Comment by lukemorawski** [comment]
@thienlnam Yes please

**Comment by thienlnam** [comment]
Code changes look good, cc @Santhosh-Sellavel now that the other issue is solved do you mind re-testing this on a simulator again to verify this works as expected?

**Comment by Santhosh-Sellavel** [comment]
Will check this again.

**Comment by Santhosh-Sellavel** [comment]
I've retested this works well. But will continue testing the complete checklist on the weekend.

https://github.com/Expensify/App/assets/85645967/edd1e2a4-29c9-4e62-a0e9-839900d7d573

**Comment by thienlnam** [comment]
Bump here - Could we please fix the conflicts and then @Santhosh-Sellavel could you review soon after?

**Review by parasharrajat — CHANGES_REQUESTED** [review]

**Comment by lukemorawski** [comment]
@parasharrajat It is suppose to behave like that. Please read the discussion from [here](https://github.com/Expensify/App/issues/19642#issuecomment-1575046405)

**Comment by lukemorawski** [comment]
@parasharrajat ok, done

**Comment by lukemorawski** [comment]
> BUG: iOS Safari: When multiline text is entered in composer, the whole composer area becomes scrollable.
> 
> Steps:
> 
> 1. Open any chat.
> 2. type 5 lines of text in composer.
> 3. Try to swipe up and down over the composer.
> 
>  Screen.Recording.2023-11-15.at.6.07.58.PM.mov

can't reproduce


https://github.com/Expensify/App/assets/134388952/0f6db6dd-d527-4ee0-a8fb-dc0388361228

**Comment by lukemorawski** [comment]
are you sure you are running the code from this PR? Cause honestly it looks like no changes to code have been made. I just run it on `main` and the app behaves exactly as on you video, while on my branch everything is OK.

**Comment by lukemorawski** [comment]
also noticed that your running it on localhost, while the new way is to run it using the `dev.new.expensify.com:8082` local domain over https. I had an issue yesterday with an app behaving weirdly over localhost(android) vs the new domain.

**Review by parasharrajat — CHANGES_REQUESTED** [review]

File: src/components/SwipeableView/index.native.tsx
Comment (lukemorawski) [reply src/components/SwipeableView/index.native.tsx:39]: great spotting! But this is deliberate :) When the style is added in native it brakes the layout

Comment (lukemorawski) [reply src/components/SwipeableView/index.native.tsx:39]: left the comment as I don't really have the capacity now to deal with that styling problem. Hopefully that's good enough.

**Review by parasharrajat — APPROVED** [review]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.


:ribbon: :eyes: :ribbon:  C+ reviewed

**Comment by lukemorawski** [comment]
on it!

File: src/libs/NativeWebKeyboard/index.ts
Comment (lukemorawski) [reply src/libs/NativeWebKeyboard/index.ts]: oops, that was mistake. Brought them back

**Review by parasharrajat — APPROVED** [review]
LGTM

**Review by thienlnam — APPROVED** [review]
Awesome, thanks for the work here!

**Pull request merged**
